### PR TITLE
deprecate AttributeMapToMap in favour of AsRaw

### PR DESCRIPTION
**Pull request by codeboten**
**Description:** 
Deprecating `AttributeMapToMap`. Moving the functionality to AttributeMap instead.

**Link to tracking Issue:** Part of #3926

### CODE REVIEW

File: model/pdata/common.go [hunk 1]
```diff
@@ -744,6 +744,31 @@ func (am AttributeMap) CopyTo(dest AttributeMap) {
 	*dest.orig = origs
 }
 
+// AsMap converts an OTLP AttributeMap to a standard go map
+func (am AttributeMap) AsMap() map[string]interface{} {
```
Comment (bogdandrutu) [inline model/pdata/common.go:748]: I was thinking that we should better return the `map[string]AttributeValue` to be consistent with New..FromMap?

Comment (codeboten) [reply model/pdata/common.go:748]: Yeah I'm working through this, the only pain point is the json marshalling.

Comment (bogdandrutu) [reply model/pdata/common.go:748]: That is true, and I have this in my mind. We have the "AsString" from AttributeValue not really producing JSON encoding, maybe we should?

Comment (bogdandrutu) [reply model/pdata/common.go:748]: It is interesting because:

```golang
// This produces "foo"
pdata.NewAttributeValueString("foo").AsString()
// This produces "{\"foo\": \"bar\"}"
attr := pdata.NewAttributeValueMap()
attrMap := attr.Map()
attrMap.UpsertString("foo", "bar")
attr.AsString()
```

So we are inconsistent on how we encode strings in these cases :(

Comment (codeboten) [reply model/pdata/common.go:748]: Yup, the inconsistency is described here: https://github.com/open-telemetry/opentelemetry-collector/issues/3815#issuecomment-896635804

Comment (codeboten) [reply model/pdata/common.go:799]: I've created a PR to see what changing the return looks like, PTAL: https://github.com/codeboten/opentelemetry-collector/pull/1

Comment (codeboten) [reply model/pdata/common.go:799]: Looking at the use of `AttributeMapToMap` in the contrib repo, it seems the best thing to do here may be to just create a `toJSON` method as it appears to be the most common usecase

Comment (codeboten) [reply model/pdata/common.go]: OK. Merged the change to update the return type, added a separate private method to ensure JSON marshalling will continue to work for AttributeMap. PTAL @bogdandrutu

**Comment by codeboten** [comment]
The test failure in contrib is unrelated to this change.

**Review by carlosalberto — APPROVED** [review]

File: model/pdata/common.go [hunk 2]
```diff
@@ -795,8 +795,25 @@ func (am AttributeMap) CopyTo(dest AttributeMap) {
...
+		return true
+	})
+	return rawMap
+}
+
 // AttributeMapToMap converts an OTLP AttributeMap to a standard go map
+// Deprecated: use AttributeMap's AsMap method instead.
 func AttributeMapToMap(attrMap AttributeMap) map[string]interface{} {
+	return attributeMapToMarshalable(attrMap)
+}
```
Comment (bogdandrutu) [inline model/pdata/common.go:812]: So what I am hearing is that it is still useful to have "map[string]interface{}" for JSON encoding. Then let's do the following:
1. Have a `func (am AttributeMap) AsMap() map[string]AttributeValue`
2. Have a `func (am AttributeMap) AsRaw() map[string]interface{}`

We can then have an equivalent `AsRaw` on AttributeValue and on ArrayAny that can be used. 

In this way we follow the pattern that `AsRaw` returns "raw" structs.

**Review by bogdandrutu — COMMENTED** [review]
@codeboten what do you think?

File: model/pdata/common.go (hunk 2 above)
Comment (codeboten) [reply model/pdata/common.go]: Implemented `AsRaw`, PTAL

File: model/pdata/common.go [hunk 3]
```diff
@@ -820,24 +830,26 @@ func AttributeMapToMap(attrMap AttributeMap) map[string]interface{} {
...
+// AttributeMapToMap converts an OTLP AttributeMap to a standard go map
+// Deprecated: use AttributeMap's AsRaw method instead.
+func AttributeMapToMap(attrMap AttributeMap) map[string]interface{} {
+	return attrMap.AsRaw()
+}
+
 // attributeArrayToSlice creates a slice out of a AnyValueArray.
-func attributeArrayToSlice(attrArray AnyValueArray) []interface{} {
-	rawSlice := make([]interface{}, 0, attrArray.Len())
+func attributeArrayToSlice(attrArray AnyValueArray) []AttributeValue {
```
Comment (Aneurysm9) [inline model/pdata/common.go:840]: Is this change necessary?  It seems like it'll result in `AttributeMap.AsRaw()` converting maps to `map[string]interface{}` while arrays will become `[]AttributeValue`.  Should they both be either `AttributeValue` or `interface{}`?

Comment (codeboten) [reply model/pdata/common.go]: Correct this is no longer necessary with the change to return a `[]interface` in `AsRaw`. Reverted the change.

**Review by Aneurysm9 — APPROVED** [review]

**Comment by alolita** [comment]
@tigrannajaryan can you please review and merge?

**Review by tigrannajaryan — APPROVED** [review]
